codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -756,7 +756,14 @@ public ChannelFuture resetStream(final ChannelHandlerContext ctx, int streamId,
             return promise.setSuccess();
         }
 
-        ChannelFuture future = frameWriter().writeRstStream(ctx, streamId, errorCode, promise);
+        final ChannelFuture future;
+        if (connection().local().created(stream) && !stream.isFrameSent()) {
```

Comment:
Do we need to add `isFrameSent()` and another method/concept? Would it be sufficient to explicitly check the state in this case (the encoder does this for most frames)?